commercial/app/model/capi/Lookup.scala

```diff
@@ -26,11 +26,35 @@ class Lookup(contentApiClient: ContentApiClient) extends Logging with implicits.
     }
   }
 
+  def shortUrlToShortId(shortUrl: String): String = {
+    /*
+        Date: 02nd Dec 2020
+        id: 288767d7-ba82-4d67-8fb3-9139e67b0f2e
+
+        CAPI (recently) announced that we would be moving
+
+        from
+          https://gu.com/p/abc
+
+        to
+          https://www.theguardian.com/p/abc
+
+        for "short" urls.
+
+        Introducing this to handle gracefully both the old and new convention without having to
+        worry when the change will actually happen. This function can be simplified in the future when the migration
+        has completed.
+     */
+    shortUrl
+      .replaceFirst("^[a-zA-Z]+://gu.com/", "")
+      .replaceFirst("^[a-zA-Z]+://www.theguardian.com/", "")
```

Comment:
Why `[a-zA-Z]+:`, and not `https?:`?

Why check for both upper and lower case in the protocol, but not the rest of the URL?

Reply:
The first one, the 

```
replaceFirst("^[a-zA-Z]+://gu.com/", "")
``` 

is the original one that I did not modify, it was already there. The second one 

```
replaceFirst("^[a-zA-Z]+://theguardian.com/", "")
```

Is the one I introduced. Both come from CAPI, and are generated lowercase :)